src/main/java/org/wildfly/security/http/impl/NonceManager.java

```diff
@@ -43,7 +43,7 @@
  *
  * @author <a href="mailto:darran.lofthouse@jboss.com">Darran Lofthouse</a>
  */
-class NonceManager {
+public class NonceManager {
```

Comment:
What is the reason for making this public?  If the only reason is for the test cases we may want to look at an alternative option.

Custom NonceManagers is something to still consider so maybe that could be addressed first before this PR and then this PR can build on that.